### ISSUE TITLE
COPY ... TO stdout docs example throws errors

### ISSUE THREAD

**Issue by bolek** [elixir-ecto/postgrex#228]
I am probably doing something terribly bad. Trying to test out `COPY *** TO STDOUT` with streams.

Following the example in the [docs](https://hexdocs.pm/postgrex/Postgrex.html#stream/4):
```elixir
Postgrex.transaction(pid, fn(conn) ->
  query = Postgrex.prepare!(conn, "COPY posts TO STDOUT")
  stream = Postgrex.stream(conn, query, [])
  Enum.into(stream, File.stream!("posts"))
end)
```

I wrote sth that corresponds to my database:

```elixir
Postgrex.transaction(pid, fn(conn) ->
  query = Postgrex.prepare!(conn, "COPY users TO STDOUT", [])
  stream = Postgrex.stream(conn, query, [])
  Enum.into(stream, File.stream!("users"))
end)
```

When I ran it in `iex`, I get the following error:
```
** (FunctionClauseError) no function clause matching in IO.binwrite/2
           (elixir) lib/io.ex:176: IO.binwrite({:file_descriptor, :prim_file, {#Port<0.15015>, 25}}, %Postgrex.Result{columns: nil, command: nil, connection_id: nil, num_rows: 0, rows: nil})
           (elixir) lib/enum.ex:1628: anonymous fn/3 in Enum.reduce/3
           (elixir) lib/stream.ex:1138: Stream.do_resource/5
           (elixir) lib/enum.ex:1627: Enum.reduce/3
           (elixir) lib/enum.ex:1121: Enum.into/4
    (db_connection) lib/db_connection.ex:1063: DBConnection.transaction_run/4
    (db_connection) lib/db_connection.ex:987: DBConnection.run_begin/3
    (db_connection) lib/db_connection.ex:667: DBConnection.transaction/3
```

It looks like the result is empty. So I tried creating the prepared query and stream outside of the transaction:

```elixir
iex(3)>   query = Postgrex.prepare!(pid, "COPY users TO STDOUT", [])
%Postgrex.Query{columns: nil, copy_data: false, decoders: nil, encoders: [],
 name: "COPY wi.users TO STDOUT", null: nil, param_formats: [], param_oids: [],
 ref: #Reference<0.0.2.1873>, result_formats: [], result_oids: nil,
 statement: [], types: 4128915}

iex(4)>   stream = Postgrex.prepare!(pid, query, [])
** (ArgumentError) argument error
                    :erlang.iolist_to_binary(%Postgrex.Query{columns: nil, copy_data: false, decoders: nil, encoders: [], name: "COPY wi.users TO STDOUT", null: nil, param_formats: [], param_oids: [], ref: #Reference<0.0.2.1873>, result_formats: [], result_oids: nil, statement: [], types: 4128915})
         (postgrex) lib/postgrex/query.ex:46: DBConnection.Query.Postgrex.Query.parse/2
    (db_connection) lib/db_connection.ex:780: DBConnection.parse/4
    (db_connection) lib/db_connection.ex:409: DBConnection.prepare/3
    (db_connection) lib/db_connection.ex:426: DBConnection.prepare!/3
```

What am I missing?

Using Elixir 1.3.2
Postgrex 0.11.2

**Comment by fishcakez** [comment]
Ah the examples are bad. I guess this would work:
```elixir
Postgrex.transaction(pid, fn(conn) ->
  query = Postgrex.prepare!(conn, "COPY users TO STDOUT", [])
  stream = Postgrex.stream(conn, query, [])
  Enum.into(stream, File.stream!("users"), fn(%Postgrex.Result{rows: rows}) -> rows end)
end)
```

**Comment by bolek** [comment]
@fishcakez - thank you!

I noticed `Postgrex.prepare` was also incorrect.

This works:

```elixir
Postgrex.transaction(pid, fn(conn) ->
  {:ok, query} = Postgrex.prepare(conn, "", "COPY users TO STDOUT")
  Postgrex.stream(conn, query, [])
    |>  Enum.into(File.stream!("users"), fn(%Postgrex.Result{rows: rows}) -> rows end)
end)
```

**Comment by fishcakez** [comment]
Fixed in e4020ff and 20cb284
